src/compiler/scala/tools/nsc/transform/patmat/MatchAnalysis.scala

```diff
@@ -374,10 +374,43 @@ trait MatchApproximation extends TreeAndTypeAnalysis with ScalaLogic with MatchT
         def tru                                               = True
       }
 
+
+      private def isIrrefutabilityProof(sym: Symbol): Boolean = {
+        sym.isMethod && sym.name == TermName("isEmpty") && {
+          //In some cases this check returns true while checking for
+          //`ConstantFalse` returns false, for example when compiling
+          //test/files/neg/t12240.scala
+          sym.asMethod.returnType match {
```

Comment:
Would / should `asMethod` work with if it's defined as `val isEmpty` (instead of `def`)? Maybe use `sym.tpe.finalResultType`?

Reply:
Looks like val defs are also method symbols?

```
scala> class Foo { val isEmpty: false = false }
class Foo

scala> :power
Power mode enabled. :phase is at typer.
import scala.tools.nsc._, intp.global._, definitions._
Try :help or completions for vals._ and power._

scala> symbolOf[Foo].tpe.member(TermName("isEmpty")).asMethod
val res0: $r.intp.global.MethodSymbol = value isEmpty
```

Reply:
That's what I expected (it uses the accessor). Add a test?

Reply:
I changed it your way :)